### PR TITLE
feat(tab-select): initial implementation for tab select *experimental*

### PR DESCRIPTION
**Pull request by emoralesb05**
## Description

First pass on tab select component and will be treated as experimental while we use it in multiple products to ensure quality.

Usage:

```html
<td-tab-select [value]="myValue"
                [backgroundColor]="'primary'"
                [color]="'accent'"
                [disabled]="false"
                [disabledRipple]="false"
                (valueChange)="myValue = $event">
  <td-tab-option [value]="1" [disabled]="false">Label 1</td-tab-option>
  <td-tab-option [value]="2">Label 2</td-tab-option>
  <td-tab-option [value]="3" [disabled]="true">Label 3</td-tab-option>
</td-tab-select>
```

### What's included?

- README.md
- CovalentTabSelectModule with TdTabSelectComponent and TdTabOptionComponent
- Usage on test bed
- Added in experimental

#### Test Steps

- [x] `npm run serve:test-bed`
- [x] See it work

#### General Tests for Every PR

- [x] `npm run serve:prod` still works.
- [x] `npm run tslint` passes.
- [x] `npm run stylelint` passes.
- [x] `npm test` passes and code coverage is not lower.
- [x] `npm run build:lib` still works.

##### Screenshots or link to StackBlitz/Plunker

image

closes https://github.com/Teradata/covalent/issues/1178

### CODE REVIEW

File: src/platform/experimental/tab-select/tab-option.component.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<ng-template>
+  <ng-content></ng-content>
+</ng-template>
```
Comment (JulieMarie) [inline src/platform/experimental/tab-select/tab-option.component.html]: Will the extra special characters cause any problems?

File: src/platform/experimental/tab-select/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
++ value?: any
+  + Sets the value of the component.
++ disabled?: boolean
+  + Sets disabled state of the component.
++ disabledRipple?: boolean
+  + Disables ripple effect on component.
++ color?: ThemePalette
+  + Color of the tab group.
++ backgroundColor: ThemePalette
```
Comment (JulieMarie) [inline src/platform/experimental/tab-select/README.md]: Is `backgroundColor` optional or required?

File: src/platform/experimental/tab-select/tab-option.component.html (hunk 1 above)
Comment (JulieMarie) [reply src/platform/experimental/tab-select/tab-option.component.html]: Maybe these aren't a problem...I see them on test-bed.component.html as well.

File: src/platform/experimental/tab-select/README.md (hunk 2 above)
Comment (emoralesb05) [reply src/platform/experimental/tab-select/README.md]: good catch

File: src/platform/experimental/tab-select/tab-option.component.html (hunk 1 above)
Comment (emoralesb05) [reply src/platform/experimental/tab-select/tab-option.component.html]: what extra special characters?

Comment (JulieMarie) [reply src/platform/experimental/tab-select/tab-option.component.html]: Weird....they are gone now.

File: src/test-bed/test-bed/test-bed.component.html [hunk 3]
```diff
@@ -1 +1,57 @@
...
+      MY LABEL
+    </td-tab-option>
+    <td-tab-option>
+      MY LABEL 2
+    </td-tab-option>
+    <td-tab-option>
+      MY LABEL 3
+    </td-tab-option>
+  </td-tab-select>
+</div>
```
Comment (JulieMarie) [inline src/test-bed/test-bed/test-bed.component.html]: But still down here. ;-)

screen shot 2018-07-06 at 2 13 03 pm

Comment (jeremysmartt) [inline src/test-bed/test-bed/test-bed.component.html]: no example with disabledRipple, do we care?

Comment (emoralesb05) [reply src/test-bed/test-bed/test-bed.component.html]: added~

**Comment by kyleledbetter** [comment]
`mat-stretch-tabs` would be a nice addition

**Comment by emoralesb05** [comment]
Will be taken care of in https://github.com/Teradata/covalent/pull/1191

**Pull request merged**